api/src/main/java/org/openmrs/module/appointments/helper/AppointmentServiceHelper.java

```diff
@@ -97,4 +102,35 @@ public void validateStatusChangeAndGetErrors(Appointment appointment,
             throw new APIException(message);
         }
     }
+
+    public List<AppointmentConflict> getConflictsForMultipleAppointments(
+            List<Appointment> appointments, List<AppointmentConflictType> appointmentConflictTypes) {
+        List<Appointment> filteredAppointments = getNonVoidedFutureAppointments(appointments);
+        List<AppointmentConflict> allConflicts = new ArrayList<>();
+        for (Appointment appointment : filteredAppointments) {
+            List<AppointmentConflict> appointmentConflicts = getConflictsForSingleAppointment(appointment, appointmentConflictTypes);
+            if (CollectionUtils.isNotEmpty(appointmentConflicts))
+                allConflicts.addAll(appointmentConflicts);
+        }
+        return allConflicts;
+    }
+
+    private List<Appointment> getNonVoidedFutureAppointments(List<Appointment> appointments) {
+        return appointments.stream()
+                .filter(appointment -> !(appointment.getVoided() || appointment.getStartDateTime().before(new Date())))
```

Comment:
Can we use isFutureAppointment in appointment.java? Also when we are comparing the date we need to compare with start of day not with time

Reply:
While checking conflicts so we include all statuses - MISSED, SCHEDULED, CANCELLED all of them?